plugin/trino-http-event-listener/src/main/java/io/trino/plugin/httpquery/HttpEventListener.java

```diff
@@ -162,6 +162,18 @@ public void onFailure(Throwable t)
                 (long) delay.getValue(), delay.getUnit());
     }
 
+    private boolean shouldRetry(StatusResponse result)
+    {
+        int statusCode = result.getStatusCode();
+        if (statusCode >= 200 && statusCode < 300) {
+            return false; // OK
+        }
+        if (statusCode >= 400 && statusCode < 500) {
```

Comment:
actually I am not convinced that we should retry anything other than 5xx, as it was originally. Is there any reason for other codes @mosiac1 .

Thanks @hashhar for pointing this out :)

Reply:
my reasoning was to help prevent dropped events - if there is no retry then that event is lost. in some systems codes other than 5xx can be resolved and are due to a middleman issue (like kubernetes or a gateway).

Reply:
408 Request Timeout and 429 Too Many Requests are the ones that I know people retry. Other 4xx seem like they'd keep failing e.g. 403, 404, 400 etc.